--- a/CMakeLists.txt
+++ b/CMakeLists.txt
@@ -167,16 +167,20 @@
   Support
 )
 if( LIBCLC_STANDALONE_BUILD )
-  add_llvm_executable( prepare_builtins utils/prepare-builtins.cpp )
-  set( prepare_builtins_exe prepare_builtins )
-  set( prepare_builtins_target prepare_builtins )
+  if( NOT prepare_builtins_exe )
+    add_llvm_executable( prepare_builtins utils/prepare-builtins.cpp )
+    set( prepare_builtins_exe prepare_builtins )
+    set( prepare_builtins_target prepare_builtins )
+  endif()
 else()
   add_llvm_utility( prepare_builtins utils/prepare-builtins.cpp )
   setup_host_tool( prepare_builtins PREPARE_BUILTINS prepare_builtins_exe prepare_builtins_target )
 endif()
-target_compile_definitions( prepare_builtins PRIVATE ${LLVM_VERSION_DEFINE} )
-# These were not properly reported in early LLVM and we don't need them
-target_compile_options( prepare_builtins PRIVATE -fno-rtti -fno-exceptions )
+if( TARGET prepare_builtins )
+  target_compile_definitions( prepare_builtins PRIVATE ${LLVM_VERSION_DEFINE} )
+  # These were not properly reported in early LLVM and we don't need them
+  target_compile_options( prepare_builtins PRIVATE -fno-rtti -fno-exceptions )
+endif()
 
 # Setup arch devices
 set( r600--_devices cedar cypress barts cayman )
